### ISSUE TITLE
Support Stryker dashboard

### ISSUE THREAD

**Issue by leobenkel** [stryker-mutator/stryker4s#208]
### Summary

I was wondering is there is any plans to have it compatible with platform like https://coveralls.io/ ? 

Thanks

**Comment by hugo-vrijswijk** [comment]
Thanks for creating this issue! We don't support coveralls, as the metrics needed are a little different. However, Stryker does have a [dashboard ](https://dashboard.stryker-mutator.io/) where you can create a badge for your README after CI jobs. In the future, we also want to add support for showing a HTML report on it. Currently, Stryker4s does not support the dashboard yet. 

Is this what you are looking for? If so, I will rename this issue to supporting the dashboard

**Comment by leobenkel** [comment]
So you are saying that from a Scala/SBT project, I am `NOT` able to use the dashboard today ?

**Comment by leobenkel** [comment]
I just signed up to the platform and it seems that I can use it the same way Coveralls works. Just have to submit the result after the coverage test using the API key. 

But there is no SBT command to upload automatically. Is that the missing link ?

**Comment by hugo-vrijswijk** [comment]
Yes. Currently the dashboard only works with Stryker for JavaScript. So not for sbt (or maven) projects. When we support it, it'll probably be integrated into the `stryker` task and upload it on CI environments after a run. Similar to Stryker for JavaScript, instead of as a separate task

**Comment by leobenkel** [comment]
There is no way to do a curl push of the report ? It the only issue is the missing push of the result, there might be a hacky way to do it for now ? Or is it a deeper format issue ? 
Thanks !

**Comment by hugo-vrijswijk** [comment]
Hmm, it should be possible, I guess? I imagine Stryker just does a POST to the dashboard. It's not an 'official' way, but you might be able to find something in the Stryker source code

**Comment by leobenkel** [comment]
So it works when you do:

```
curl -X POST \
  https://dashboard.stryker-mutator.io/api/reports \
  -H 'Content-Type: application/json' \
  -d '{
    "apiKey": "[API_KEY]",
    "repositorySlug": "github.com/[USER]/[REPO_NAME]",
    "branch": "[BRANCH]",
    "mutationScore": [MUTATION_SCORE]
}'
```

**Comment by leobenkel** [comment]
Now I need to find a way to read the actual percent...

**Comment by leobenkel** [comment]
Is there a reporter that would dump a json file maybe where I could read the final score?

**Comment by hugo-vrijswijk** [comment]
The html reporter writes to a .js file with a JSON of the report results. That doesn't have a score though, only the killed/survived amounts. Reading that file, calculating the score and POSTing it is probably a similar amount of work as making a dashboard reporter in Stryker4s

**Comment by leobenkel** [comment]
yeah, I looked at the js file and it doesnt look easy to parse haha !

**Comment by leobenkel** [comment]
https://github.com/stryker-mutator/stryker4s/pull/209 I made a quick Json reporter to be able to move forward with bash commands. Fetching the branch name and the repo slug from within the code won't be trivial

**Comment by legopiraat** [comment]
Closing fixed with #210
